### DISCUSSION TITLE
Spin state energies for bimetallic complexes using already saved integrals

### DISCUSSION THREAD

**Discussion by arnab82** [block-hczhai/block2-preview#119] [Q&A] [answered]
Dear Block2 developers,
First of all, thank you for developing such an excellent program and providing comprehensive documentation!
I am trying to compute spin-state energies of a bimetallic complex (Cr2O(NH3)_10) using my own ROHF integrals. I am doing state-average DMRG calculation and not sure is this a good way to compute the spin-state energies for this complex. Although ST-DMRG would be a great option for this complex, but I don't want to use downfolded hamiltonian as I want to see how these integrals perform for DMRG calculation. Please see below

`import numpy as np
ncore=67 
ncas=29
n_elec=26 #fockspace=(6,3),(4,4),(6,3)
spin=6
charge=4

orb_sym=[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0,0,0,0,0,0,0,0,0,0,0,0,0]
h1e=np.load("./ints_h1.npy")
g2e=np.load("./ints_h2.npy")
ecore=np.load("./ints_h0.npy")

driver = DMRGDriver(scratch="./tmp", symm_type=SymmetryTypes.SZ, n_threads=4)
#driver = DMRGDriver(scratch="./tmp", symm_type=SymmetryTypes.SU2, n_threads=4)
idx = driver.orbital_reordering(np.abs(h1e), np.abs(g2e))
print('reordering = ', idx)
h1e = h1e[idx][:, idx]
g2e = g2e[idx][:, idx][:, :, idx][:, :, :, idx]
driver.initialize_system(n_sites=ncas, n_elec=n_elec, spin=spin, orb_sym=orb_sym)

mpo = driver.get_qc_mpo(h1e=h1e, g2e=g2e, ecore=ecore, iprint=0)

ket = driver.get_random_mps(tag="KET", bond_dim=250, nroots=4)
bond_dims = [400] * 4 + [800] * 8
noises = [1e-5] * 4 + [1e-6] * 4 + [0]
thrds = [1e-7] * 4 + [1e-9] * 4
energies = driver.dmrg(mpo, ket, n_sweeps=32, bond_dims=bond_dims, noises=noises,
    thrds=thrds, iprint=1)
print('State-averaged MPS energies = [%s]' % " ".join("%20.15f" % x for x in energies))

kets = [driver.split_mps(ket, ir, tag="KET-%d" % ir) for ir in range(ket.nroots)]
for ir in range(ket.nroots):
    energy = driver.dmrg(mpo, kets[ir], n_sweeps=32, bond_dims=bond_dims, noises=noises,
        thrds=thrds, iprint=0, proj_weights=[5.0] * ir, proj_mpss=kets[:ir])
    print('State-specific MPS E[%d] = %20.15f' % (ir, energy))`

**Comment by hczhai** [comment, marked as answer]
Hi, thanks for your interest in using the block2 package. The accuracy of DMRG energies can be evaluated by using the discarded weights and the distance to the extrapolated energy, following the procedure described in https://block2.readthedocs.io/en/latest/tutorial/energy-extrapolation.html.

**Reply by arnab82** [reply]
Thank you for the response. This helped me a lot.
